core/src/commonMain/kotlin/com/sourcepoint/mobile_core/models/SPPropertyName.kt

```diff
@@ -1,4 +1,55 @@
 package com.sourcepoint.mobile_core.models
 
-// TODO validate property name
-typealias SPPropertyName = String
+import kotlinx.serialization.*
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.descriptors.*
+
+@Serializable(with = SPPropertyName.Serializer::class)
+open class SPPropertyName private constructor(val rawValue: String) {
+    override fun toString(): String = rawValue.removePrefix("https://")
+
+    companion object {
+        private val VALID_REGEX = Regex("^[a-zA-Z.:/0-9-]*$")
+
+        @Throws(InvalidPropertyNameError::class)
+        fun create(rawValue: String): SPPropertyName {
+            val validatedValue = validate(rawValue)
+            val finalValue = if (!validatedValue.startsWith("https://") && !validatedValue.startsWith("http://")) {
+                "https://$validatedValue"
+            } else {
+                validatedValue
+            }
+            return SPPropertyName(finalValue)
+        }
+
+        @Throws(InvalidPropertyNameError::class)
+        private fun validate(string: String): String {
+            if (!VALID_REGEX.matches(string)) {
+                throw InvalidPropertyNameError("PropertyName can only include letters, numbers, '.', ':', '-' and '/'. '$string' is invalid")
+            }
+            return string
+        }
+    }
+
+    override fun equals(other: Any?): Boolean =
+        other is SPPropertyName && rawValue == other.rawValue
+
+    override fun hashCode(): Int = rawValue.hashCode()
+
+    object Serializer : KSerializer<SPPropertyName> {
+        override val descriptor: SerialDescriptor = PrimitiveSerialDescriptor(
+            serialName = "SPPropertyName",
+            kind = PrimitiveKind.STRING
+        )
+
+        override fun serialize(encoder: Encoder, value: SPPropertyName) {
+            encoder.encodeString(value.rawValue)
+        }
+
+        override fun deserialize(decoder: Decoder): SPPropertyName =
+            create(decoder.decodeString())
+    }
+}
+
+class InvalidPropertyNameError(message: String) : IllegalArgumentException(message)
```

Comment:
@andresilveirah I suggest moving the exception class to the `com/sourcepoint/mobile_core/models/SPError.kt` to have all of them in the same place.

Reply:
Also, to conform iOS policy of error codes I'd go with something like this:
```
open class InvalidPropertyNameError(
    propertyName: String
): SPError (
    code = "sp_metric_invalid_property_name",
    description = "PropertyName can only include letters, numbers, '.', ':', '-' and '/'.  ${propertyName} passed is invalid"
)

```